### ISSUE TITLE
Differentiable Function implementations for appropriate numpy/pycuda methods

### ISSUE THREAD

**Issue by EdwardRaff** [chainer/chainer#814]
It would be nice if there were more differentiable functions already implemented that simply call their respective numpy/pycuda counterparts (i.e: np.squared, sin/cos/tan/etc, prod, expm1, logaddexp, clip, sqrt). This would make it easier to exploit the auto-diff abilities of Chainer to implement our own custom loss functions without having to implement the backprop code.

**Comment by beam2d** [comment]
Many functions are added since then. Make a new issue if you need some that are still lacking.
